src/client_state.cpp

```diff
@@ -86,7 +87,7 @@ int wsrep::client_state::before_command()
     if (transaction_.active() &&
         server_state_.rollback_mode() == wsrep::server_state::rm_sync)
     {
-        while (transaction_.state() == wsrep::transaction::s_aborting)
```

Comment:
The old code is checking first for transaction state to be s_aborting, and only then goes to waiting for the cond_ signal.
But the rollbacker will first set the victim transaction's state to s_aborted when rollback is complete in after_rollback() hook. If the victim wakes up at this point and checks if it can enter before_command(), it will go through. But, note that rolllbacker has not yet finished yet, it will continue using the client's state and THD, and will leave the client in sync_rollback_complete(), and there it sill send the cond_ signal, which client should have waited for.